### PR TITLE
EN-2033 [Node.js SDK] Do not throw error if can't parse/decrypt record in Find results

### PR DESCRIPTION
**Pull request by vladimir-grezin**
Pull Request Checklist
======================

Please read this for our PR Culture:
------------------------------------
https://incountry.atlassian.net/wiki/spaces/ED/pages/46858869/Pull+requests+culture+Draft

Work relates to:
----------------
Link(s) to Jira Task/User Story/Bug/Epic:
- https://incountry.atlassian.net/browse/EN-2033

Code Quality (Before submitting PR)
------------
- [x] Ensure code compiles and passes Sonar Lint.
- [x] Do your best to ensure integration tests pass. If they don't then...
- [x] Remove, ignore, or mark pending any brittle tests.
- [x] Remove commented out code.

Target Release
--------------
- [ ] Please put the target release vehicle this needs to be bundled with.
- [ ] Did you increment the version in package.json? What is the new version?

### CODE REVIEW

File: storage.js [hunk 1]
```diff
@@ -146,10 +146,25 @@ class Storage {
...
+      result.data = [];
+      const decrypted = await Promise.all(
+        response.data.data.map((item) => this._decryptPayload(item).catch((e) => ({
+          error: e.message,
+          rawData: item,
+        }))),
+      );
+      decrypted.forEach((item) => {
+        if (item.error) {
+          result.errors = result.errors ? [...result.errors, item] : [item];
```
Comment (igor-alekseev-dev) [inline storage.js]: @vladimir-grezin why not create `errors: []` beforehand?

File: storage.js [hunk 2]
```diff
@@ -146,10 +146,25 @@ class Storage {
       data,
     });
     if (response.data) {
-      const decryptedData = await Promise.all(response.data.data.map((item) => this._decryptPayload(item)));
+      const result = {};
```
Comment (RommeroInCountry) [inline storage.js]: It'd be better to initialise result as 
`
const results = { data: [], errors: [] } 
`

File: tests/unit/storage-tests.js [hunk 3]
```diff
@@ -200,6 +200,46 @@ describe('Storage', function () {
...
+    const unsupportedData = {
+      country: 'us',
+      key: 'somekey',
+      body: 'unsupported data'
+    };
+
+    const data = [...encryptedData, unsupportedData]
+    nock('https://us.api.incountry.io')
+      .post(`/v2/storage/records/us/find`)
+      .reply(200, (uri, requestBody) => {
```
Comment (artemgolozubov) [inline tests/unit/storage-tests.js]: I think we can simplify this reply with a plain object, like this
```
.reply(200, { meta: { total: data.length }, data });
```

File: tests/unit/storage-tests.js [hunk 4]
```diff
@@ -200,6 +200,46 @@ describe('Storage', function () {
...
+      TEST_RECORDS.map((record) => encryptedStorage._encryptPayload(record))
+    );
+
+    const unsupportedData = {
+      country: 'us',
+      key: 'somekey',
+      body: 'unsupported data'
+    };
+
+    const data = [...encryptedData, unsupportedData]
```
Comment (artemgolozubov) [inline tests/unit/storage-tests.js]: Looks like there are several linting issues in this file (like missing semicolon in the end of this line or missing spaces in other lines). I think it'd be a good idea to check all changes in the old code with eslint (despite disabled eslint for this file). What do you think about it?

File: storage.js (hunk 1 above)
Comment (vladimir-grezin) [reply storage.js]: Because we don't want (I suppose) to have `errors` in case if everything's fine

File: tests/unit/storage-tests.js (hunk 3 above)
Comment (vladimir-grezin) [reply tests/unit/storage-tests.js]: +

File: tests/unit/storage-tests.js (hunk 4 above)
Comment (vladimir-grezin) [reply tests/unit/storage-tests.js]: I have no idea why it was disabled

File: storage.js (hunk 2 above)
Comment (vladimir-grezin) [reply storage.js]: We only want to have errors array if there are any errors

**Pull request merged**